login_server/player/state/authenticated_state.py

```diff
@@ -253,18 +271,20 @@ def _send_control_message(self, player, message: Message):
     def handle_promotion_code_redemption(self, request):
         promotion_code = request.findbytype(m0669)
         if promotion_code:
-            authcode = promotion_code.value
-            if (self.player.login_name in self.player.login_server.accounts and
-                    self.player.login_server.accounts[self.player.login_name].authcode == authcode):
+            self._handle_verification_code(promotion_code.value)
 
-                self.player.login_server.accounts[self.player.login_name].password_hash = self.player.password_hash
-                self.player.login_server.accounts[self.player.login_name].authcode = None
-                self.player.login_server.accounts.save()
-            else:
-                invalid_code_msg = a0175()
-                invalid_code_msg.findbytype(m02fc).set(STDMSG_NOT_A_VALID_PROMOTION_CODE)  # message type
-                invalid_code_msg.findbytype(m0669).set(authcode)
-                self.player.send(invalid_code_msg)
+    def _handle_verification_code(self, authcode):
```

Comment:
Why did you do this? To avoid the indentation?

Reply:
Was refactored when I was trying out sending control messages over the promotion code messages. Could be reverted although it doesn't hurt readability IMO.

Reply:
Ok never mind.